src/Microsoft.ML.Tokenizers/Model/BPE.cs

```diff
@@ -222,8 +222,9 @@ public override IReadOnlyList<Token> Tokenize(string sequence, bool isSpecialTok
         /// </summary>
         /// <param name="id">The Id to map to the token.</param>
         /// <param name="skipSpecialTokens">Indicate if want to skip the special tokens during the decoding.</param>
+        /// <param name="filterUnsupportedChars">Indicate if want to filter the unsupported characters during the decoding.</param>
         /// <returns>The mapped token of the Id.</returns>
-        public override string? IdToToken(int id, bool skipSpecialTokens = false)
+        public override string? IdToToken(int id, bool skipSpecialTokens = false, bool filterUnsupportedChars = true)
```

Comment:
Seems like we're needing to add different special cases to handle different models. Are we going to find ourselves with another model that also requires special-casing? Does this filtering need to be on a call-by-call basis, or can it instead be baked into the model via its constructor? And if it has to be call-by-call, should we start thinking about some kind of options bag that would be passed in, such that we can add more options in the future without needing to break the abstraction?

Reply:
I like your suggestion to have it in the model constructor instead of the abstraction. I think this should work nicely without committing to this in the abstraction. I applied your suggestion. Thanks!